flexget/plugins/input/npo_watchlist.py

```diff
@@ -95,173 +105,140 @@ def _login(self, task, config):
+                   'X-XSRF-TOKEN': requests.cookies['XSRF-TOKEN'],
+                   'X-Requested-With': 'XMLHttpRequest'}
+        if page > 1:
+            headers['Referer'] = episode_tiles_url.format(mediaId, page-1)  # referer from prev page
 
-                series_info = self._parse_series_info(task, config, page, series_url)
-                self._series_info[series_url] = series_info
+        log.info('Retrieving episodes page %s for %s (%s)', page, series_info['npo_name'], mediaId)
+        entries = list()
+        try:
+            episodes = requests.get(episode_tiles_url.format(mediaId, page) + episode_tiles_parameters,
+                                    headers=headers).json()
+            entries += self._parse_tiles(task, config, episodes['tiles'], series_info)
 
-            return self._series_info[series_url]
+            if episodes['nextLink']:
+                entries += self._get_series_episodes(task, config, mediaId, series_info,
+                                                     page=int(episodes['nextLink'].rsplit('page=')[1]))
         except RequestException as e:
             raise plugin.PluginError('Request error: %s' % e.args[0])
 
-    def _parse_series_info(self, task, config, page, series_url):
-        tvseries = page.find('section', class_='npo-header-episode-meta')
-        series_info = Entry()  # create a stub to store the common values for all episodes of this series
-        series_info['npo_url'] = series_url
-        series_info['npo_name'] = tvseries.find('h1').text
-        series_info['npo_description'] = tvseries.find('div', id='metaContent').find('p').text
-        series_info['npo_language'] = 'nl'  # hard-code the language as if in NL, for loookup plugins
-        return series_info
+        if not entries:
+            log.warning('No episodes found for %s (%s)', series_info['npo_name'], mediaId)
+        return entries
 
-    def _get_series_episodes(self, task, config, series_name, series_url):
-        log.info('Retrieving new episodes for %s', series_name)
+    def _get_series_info(self, task, config, mediaId):
+        series_info_url = 'https://www.npostart.nl/{0}'
+        log.info('Retrieving series info for %s', mediaId)
         try:
-            response = requests.get(series_url)
+            response = requests.get(series_info_url.format(mediaId))
             page = get_soup(response.content)
-
-            series_info = self._parse_series_info(task, config, page, series_url)
-            episodes = page.find('div', id='component-grid-episodes')
-            entries = self._parse_tiles(task, config, episodes, series_info)
-
-            if not entries:
-                log.warning('No episodes found for %s', series_name)
-
-            return entries
+            series = page.find('section', class_='npo-header-episode-meta')
+            series_info = Entry()  # create a stub to store the common values for all episodes of this series
```

Comment:
Why not just create a dict? Doesn't seem like it needs to be an entry.

Reply:
in the original code it was also like this; I will update it